sunpy/map/maputils.py

```diff
@@ -1,7 +1,252 @@
+    then the function returns True. Otherwise, the function
+    returns False.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The input map.
+
+    Returns
+    -------
+    is_all_off_disk : `~bool`
+        Returns True if all map pixels are strictly more than one solar radius
+        away from the center of the Sun.
+    """
+    return np.all(~coordinate_is_on_disk(all_coordinates_from_map(smap), smap.rsun_obs))
+
+
+def is_all_on_disk(smap):
+    """
+    Checks if the entire map is on the solar disk.
+
+    The check is performed by calculating the distance of every pixel
+    from the center of the Sun.  If they are all on-disk, then the
+    function returns True. Otherwise, the function returns False.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The input map.
+
+    Returns
+    -------
+    is_all_on_disk : `~bool`
+        Returns True if all map pixels are strictly less than one solar radius
+        away from the center of the Sun.
+    """
+    return np.all(coordinate_is_on_disk(all_coordinates_from_map(smap), smap.rsun_obs))
+
+
+def contains_limb(smap):
+    """
+    Checks if a map contains a portion of the solar limb.
+
+    The check is performed by calculating the distance of every pixel from
+    the center of the Sun.  If at least one pixel is on disk and at least one
+    pixel is off disk, the function returns True.  Otherwise, the function
+    returns False. Note that this function will also true if the entire solar
+    limb is within the field of view of the map.  Note also that in the case
+    of coronagraph images the limb itself need not be observed.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The input map.
+
+    Returns
+    -------
+    contains_limb : `~bool`
+        Returns True If at least one pixel is on disk and at least one pixel
+        is off disk.
+    """
+    on_disk = coordinate_is_on_disk(all_coordinates_from_map(smap), smap.rsun_obs)
+    return np.logical_and(np.any(on_disk), np.any(~on_disk))
+
+
+@u.quantity_input
+def coordinate_is_on_disk(coordinate, scale: u.arcsecond):
+    """
+    Checks if the helioprojective Cartesian coordinate is on disk.
+
+    The check is performed by comparing the coordinate's distance
+    from the center of the Sun to the solar radius.
+
+    Parameters
+    ----------
+    coordinate : `~astropy.coordinates.SkyCoord`, `~sunpy.coordinates.frames.Helioprojective`
+        The input coordinate.  The frame property of the input
+        coordinate must be helioprojective.
+
+    scale : `~astropy.units.Quantity`
+        The pixel scale size in units of arcseconds.
+
+    Returns
+    -------
+    is_on_disk : `~bool`
+        Returns True if the coordinate is on disk, False otherwise.
+    """
+    # Calculate the radii of every pixel in helioprojective Cartesian
+    # co-ordinate distance units.
+    if not isinstance(coordinate.frame, Helioprojective):
```

Comment:
If this is to take a SkyCoord or a frame class then we need a check for SkyCoord:
```suggestion
    if isinstance(coordinate, SkyCoord):
        coordinate = coordinate.frame
    if not isinstance(coordinate, Helioprojective):
```